xarray/core/variable.py

```diff
@@ -410,7 +410,11 @@ def copy(self, deep=True):
         If `deep=True`, the data array is loaded into memory and copied onto
         the new object. Dimensions, attributes and encodings are always copied.
         """
-        if deep and not isinstance(self.data, dask_array_type):
+        if isinstance(self._data, PandasIndexAdapter):
+            # there is no need to copy the index values here even if deep=True
+            # since pandas.Index objects are immutable
+            data = PandasIndexAdapter(self) if deep else self._data
```

Comment:
I saw this in `Coordinate.copy()` so I thought there might have some wrapper data to copy in `PandasIndexAdapter` but I admit I didn't checked it.

Indeed, now that I see why we have this in `Coordinate.copy()` I don't see any reason to have these two different code paths here.